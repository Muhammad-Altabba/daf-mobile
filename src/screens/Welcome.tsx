/**
 * Serto Mobile App
 *
 */

import * as React from "react";
import { StyleSheet, Text, View, TouchableOpacity } from "react-native";
import Config from "react-native-config";
import Analytics from "appcenter-analytics";
import codePush from "react-native-code-push";

Analytics.setEnabled(true);

interface Props {}
interface State {
  updateMetadata: string | null;
}

export default class App extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      updateMetadata: ""
    };
  }
  onButtonPress() {
    codePush.sync({
      updateDialog: {
        appendReleaseDescription: true
      },
      installMode: codePush.InstallMode.IMMEDIATE
    });
  }

  componentDidMount() {
    codePush.getUpdateMetadata().then(updateMetadata => {
      this.setState({ updateMetadata: updateMetadata && updateMetadata.label });
    });
  }

  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>Welcome to updated2 Serto!</Text>
        <Text style={styles.welcome}>Environment = {Config.ENV}</Text>
        <TouchableOpacity
          onPress={() => {
            throw new Error("Sample error nr2");
          }}
        >
          <Text style={styles.welcome}>Press here to crash the app</Text>
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => {
            Analytics.trackEvent("Sample event");
          }}
        >
          <Text style={styles.welcome}>Press here to trigger sample event</Text>
        </TouchableOpacity>
        <TouchableOpacity onPress={this.onButtonPress}>
          <Text>Check for updates</Text>
        </TouchableOpacity>
        <Text>Update: {this.state.updateMetadata}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "#F5FCFF"
  },
  welcome: {
    fontSize: 20,
    textAlign: "center",
    margin: 10,
    color: Config.BRAND_COLOR
  }
});
