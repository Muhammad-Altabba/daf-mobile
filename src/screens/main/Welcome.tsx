/**
 * Serto Mobile App
 *
 */
import React from 'react'
import {
  Container,
  Text,
  FabButton,
  Constants,
  Screen,
  Credential,
} from '@kancha/kancha-ui'
import { NavigationScreen } from '../../navigators'
import { Icons } from '../../theme'

const sertoVerifiableCredential = {
  iss: 'Serto Identity Platform',
  sub: 'did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74',
  type: 'Serto ID',
  iat: 1562769371,
  exp: 1579478400,
  claim: {
    'Serto ID': {
      name: 'Sarah Adamson',
      dateOfBirth: '22-01-75',
      country: 'USA',
      children: [
        {
          name: 'Bob',
          age: 4,
        },
        {
          name: 'Alice',
          age: 9,
        },
      ],
    },
  },
  vc: [],
}

const bankVerifiableCredential = {
  iss: 'Deutsche Bank',
  sub: 'did:ethr:0xf3beac30c498d9e26865f34fcaa57dbb935b0d74',
  type: 'Credit Worthy',
  iat: 1562769371,
  exp: 1579478400,
  claim: {
    'Credit Worthy': {
      name: 'Alice Chainy',
      dateOfBirth: '22-01-75',
      country: 'China',
      approvedLimit: 30000000,
    },
  },
  vc: [],
}

// tslint:disable-next-line:no-var-requires
const avatar1 = require('../../assets/images/space-x-logo.jpg')

export default (props: NavigationScreen) => {
  const credentials = [
    {
      title: sertoVerifiableCredential.type,
      issuer: sertoVerifiableCredential.iss,
      logo: avatar1,
      onPress: () =>
        props.navigation.navigate('Credential', {
          vc: sertoVerifiableCredential,
        }),
    },
    {
      title: bankVerifiableCredential.type,
      issuer: bankVerifiableCredential.iss,
      logo: avatar1,
      onPress: () =>
        props.navigation.navigate('Credential', {
          vc: bankVerifiableCredential,
        }),
    },
  ]
  const loadRequest = (requestData: any) => {
    setTimeout(() => {
      if (requestData.type === 'DISCLOSURE') {
        props.navigation.push('Request', {
          requestData,
        })
      }
    }, 500)
  }

  return (
    <Screen
      scrollEnabled
      fabButton={
        <Container>
          <FabButton
            testID={'SCAN_BTN'}
            onPress={() =>
              props.navigation.navigate('Scanner', {
                loadRequest,
              })
            }
            icon={Icons.SCAN}
          />
        </Container>
      }
    >
      <Container padding>
        <Container marginBottom>
          <Text type={Constants.TextTypes.H2} bold>
            Credentials
          </Text>
        </Container>
        <Container>
          {credentials.map((credential, index) => {
            return <Credential key={index} {...credential} />
          })}
        </Container>
      </Container>
    </Screen>
  )
}
