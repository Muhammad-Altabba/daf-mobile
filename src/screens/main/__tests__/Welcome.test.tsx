import 'react-native'
import React from 'react'
import Welcome from '../Welcome'
import { render, fireEvent } from 'react-native-testing-library'

jest.useFakeTimers()
jest.runAllTimers()

const navigation = {
  navigate: jest.fn(),
}

describe('Welcome', () => {
  it('renders correctly', () => {
    //@ts-ignore
    const tree = render(<Welcome navigation={navigation} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('shows the credentials and you can tap them', () => {
    //@ts-ignore
    const { getByText } = render(<Welcome navigation={navigation} />)

    expect(getByText(/Credit Worthy/i)).toBeDefined()
    fireEvent.press(getByText(/Credit Worthy/i))
    expect(navigation.navigate).toBeCalled()

    expect(getByText(/Serto Identity Platform/i)).toBeDefined()
    fireEvent.press(getByText(/Serto Identity Platform/i))
    expect(navigation.navigate).toBeCalled()
  })
})
