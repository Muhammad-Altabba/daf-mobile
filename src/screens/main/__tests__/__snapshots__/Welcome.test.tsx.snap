// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome renders correctly 1`] = `
Array [
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    contentInsetAdjustmentBehavior="never"
    getTextInputRefs={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onScroll={[Function]}
    scrollEventThrottle={200}
    scrollToBottomOnKBShow={false}
    scrollToInputAdditionalOffset={75}
    startScrolledToBottom={false}
    style={
      Object {
        "backgroundColor": "#EFEEF3",
      }
    }
  >
    <View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": undefined,
            "borderRadius": undefined,
            "flex": 1,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": 15,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": 15,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#333333",
                  "fontSize": 24,
                  "fontWeight": "bold",
                }
              }
            >
              Credentials
            </Text>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": undefined,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": "#ffffff",
                    "borderRadius": 5,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "margin": undefined,
                    "marginBottom": 15,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": 15,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "shadowColor": "#ffffff",
                    "shadowOpacity": 0.2,
                    "shadowRadius": 0.5,
                    "width": undefined,
                  }
                }
              >
                <View
                  pointerEvents="auto"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderRadius": undefined,
                      "flex": undefined,
                      "flexDirection": "row",
                      "height": undefined,
                      "justifyContent": undefined,
                      "margin": undefined,
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                      "padding": undefined,
                      "paddingBottom": undefined,
                      "paddingHorizontal": undefined,
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                      "paddingTop": undefined,
                      "width": undefined,
                    }
                  }
                >
                  <View
                    pointerEvents="auto"
                    style={
                      Object {
                        "alignItems": undefined,
                        "backgroundColor": "#EAEAEA",
                        "borderRadius": undefined,
                        "flex": undefined,
                        "flexDirection": undefined,
                        "height": undefined,
                        "justifyContent": undefined,
                        "margin": undefined,
                        "marginBottom": undefined,
                        "marginLeft": undefined,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "padding": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "width": undefined,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../src/assets/images/space-x-logo.jpg",
                        }
                      }
                      style={
                        Object {
                          "backgroundColor": "#0078ff",
                          "borderRadius": 5,
                          "height": 40,
                          "width": 40,
                        }
                      }
                    />
                  </View>
                  <View
                    pointerEvents="auto"
                    style={
                      Object {
                        "alignItems": undefined,
                        "backgroundColor": undefined,
                        "borderRadius": undefined,
                        "flex": 1,
                        "flexDirection": undefined,
                        "height": undefined,
                        "justifyContent": undefined,
                        "margin": undefined,
                        "marginBottom": undefined,
                        "marginLeft": 15,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "padding": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "width": undefined,
                      }
                    }
                  >
                    <View
                      pointerEvents="auto"
                      style={
                        Object {
                          "alignItems": undefined,
                          "backgroundColor": undefined,
                          "borderRadius": undefined,
                          "flex": undefined,
                          "flexDirection": undefined,
                          "height": undefined,
                          "justifyContent": undefined,
                          "margin": undefined,
                          "marginBottom": undefined,
                          "marginLeft": undefined,
                          "marginRight": undefined,
                          "marginTop": undefined,
                          "padding": undefined,
                          "paddingBottom": 3,
                          "paddingHorizontal": undefined,
                          "paddingLeft": undefined,
                          "paddingRight": undefined,
                          "paddingTop": undefined,
                          "width": undefined,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#333333",
                            "fontSize": 18,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Serto ID
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "color": "#BBBBBB",
                          "fontSize": 14,
                        }
                      }
                    >
                      Serto Identity Platform
                    </Text>
                  </View>
                </View>
              </View>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": "#ffffff",
                    "borderRadius": 5,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "margin": undefined,
                    "marginBottom": 15,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": 15,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "shadowColor": "#ffffff",
                    "shadowOpacity": 0.2,
                    "shadowRadius": 0.5,
                    "width": undefined,
                  }
                }
              >
                <View
                  pointerEvents="auto"
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderRadius": undefined,
                      "flex": undefined,
                      "flexDirection": "row",
                      "height": undefined,
                      "justifyContent": undefined,
                      "margin": undefined,
                      "marginBottom": undefined,
                      "marginLeft": undefined,
                      "marginRight": undefined,
                      "marginTop": undefined,
                      "padding": undefined,
                      "paddingBottom": undefined,
                      "paddingHorizontal": undefined,
                      "paddingLeft": undefined,
                      "paddingRight": undefined,
                      "paddingTop": undefined,
                      "width": undefined,
                    }
                  }
                >
                  <View
                    pointerEvents="auto"
                    style={
                      Object {
                        "alignItems": undefined,
                        "backgroundColor": "#EAEAEA",
                        "borderRadius": undefined,
                        "flex": undefined,
                        "flexDirection": undefined,
                        "height": undefined,
                        "justifyContent": undefined,
                        "margin": undefined,
                        "marginBottom": undefined,
                        "marginLeft": undefined,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "padding": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "width": undefined,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "testUri": "../../../src/assets/images/space-x-logo.jpg",
                        }
                      }
                      style={
                        Object {
                          "backgroundColor": "#0078ff",
                          "borderRadius": 5,
                          "height": 40,
                          "width": 40,
                        }
                      }
                    />
                  </View>
                  <View
                    pointerEvents="auto"
                    style={
                      Object {
                        "alignItems": undefined,
                        "backgroundColor": undefined,
                        "borderRadius": undefined,
                        "flex": 1,
                        "flexDirection": undefined,
                        "height": undefined,
                        "justifyContent": undefined,
                        "margin": undefined,
                        "marginBottom": undefined,
                        "marginLeft": 15,
                        "marginRight": undefined,
                        "marginTop": undefined,
                        "padding": undefined,
                        "paddingBottom": undefined,
                        "paddingHorizontal": undefined,
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                        "paddingTop": undefined,
                        "width": undefined,
                      }
                    }
                  >
                    <View
                      pointerEvents="auto"
                      style={
                        Object {
                          "alignItems": undefined,
                          "backgroundColor": undefined,
                          "borderRadius": undefined,
                          "flex": undefined,
                          "flexDirection": undefined,
                          "height": undefined,
                          "justifyContent": undefined,
                          "margin": undefined,
                          "marginBottom": undefined,
                          "marginLeft": undefined,
                          "marginRight": undefined,
                          "marginTop": undefined,
                          "padding": undefined,
                          "paddingBottom": 3,
                          "paddingHorizontal": undefined,
                          "paddingLeft": undefined,
                          "paddingRight": undefined,
                          "paddingTop": undefined,
                          "width": undefined,
                        }
                      }
                    >
                      <Text
                        style={
                          Object {
                            "color": "#333333",
                            "fontSize": 18,
                            "fontWeight": "bold",
                          }
                        }
                      >
                        Credit Worthy
                      </Text>
                    </View>
                    <Text
                      style={
                        Object {
                          "color": "#BBBBBB",
                          "fontSize": 14,
                        }
                      }
                    >
                      Deutsche Bank
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "bottom": 40,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        accessibilityLabel="SCAN_BTN"
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0078ff",
            "borderRadius": 50,
            "elevation": 3,
            "height": 70,
            "justifyContent": "center",
            "opacity": 1,
            "shadowColor": "#000000",
            "shadowOpacity": 0.3,
            "shadowRadius": 20,
            "width": 70,
          }
        }
        testID="SCAN_BTN"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 35,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>,
]
`;
